File: functions/src/myFunc/utility/emailEncode.ts
export function sanitizeEmail(email: string): string {
  return email.replace(/\./g, "__dot__").replace(/@/g, "__at__");
}

export function restoreEmail(safeEmail: string): string {
  return safeEmail.replace(/__at__/g, "@").replace(/__dot__/g, ".");
}

File: app/src/main/java/gaku/original/myapplication/utility/EmailEncode.kt
package gaku.original.myapplication.utility

/**
 * Emailの@や.を置換する
 */
fun sanitizeEmail(email: String): String {
    return email.replace(".", "__dot__").replace("@", "__at__")
}

fun restoreEmail(safeEmail: String): String {
    return safeEmail.replace("__at__", "@").replace("__dot__", ".")
}